### PR TITLE
Linked: Fix crashes

### PR DESCRIPTION
**Pull request by KrisXV**

### CODE REVIEW

File: mods/linked/moves.js [hunk 1]
```diff
@@ -271,11 +271,10 @@ exports.BattleMovedex = {
...
 				switch (this.effectData.turnsActivated[this.turn]) {
 				case 1: {
-					if (!this.willMove(pokemon)) {
-						for (const source of this.effectData.sources) {
-							let pseudoDecision = {choice: 'move', move: this.effectData.move[1], targetLoc: this.getTargetLoc(pokemon, source), pokemon: this.willMove(pokemon).pokemon, targetPosition: this.willMove(pokemon).targetPosition, targetSide: this.willMove(pokemon).targetSide};
-							this.queue.unshift(pseudoDecision);
-						}
+					let curAction = this.willMove(pokemon);
+					if (!curAction) {
+						let pseudoAction = {choice: 'move', move: this.effectData.move[1], targetLoc: curAction.targetLoc, pokemon: curAction.pokemon, targetPosition: curAction.targetPosition, targetSide: curAction.targetSide};
```
Comment (Marty-D) [inline mods/linked/moves.js]: How does this work? As far as I can tell, if `curAction` is falsy it's not going to be returning a `MoveAction` or `SwitchAction`.

Comment (KrisXV) [reply mods/linked/moves.js]: honestly i just kept it when i updated the code, i was too afraid to remove something too significant in the chance that it required rewriting everything else

Comment (MacChaeger) [reply mods/linked/moves.js]: How could curAction be falsy and have properties? Won't this crash?

Comment (Marty-D) [reply mods/linked/moves.js]: That's what I'm saying; the previous code would have crashed too if it weren't for the line immediately before it crashing first. `this.willMove(pokemon)` had to be false or null just to get to this point.

**Comment by KrisXV** [comment]
what?????? no??

**Comment by KrisXV** [comment]
also removed @Marty-D now is it possible to get this up asap so ppl can stop abusing the bugs and crashes

**Comment by Slayer95** [comment]
I'm afraid Linked has always purposely went against the type definitions in the battle engine ¯\\_(ツ)_/¯

File: mods/linked/moves.js (hunk 1 above)
Comment (Slayer95) [reply mods/linked/moves.js]: As you can see in the [XY code](https://github.com/Zarel/Pokemon-Showdown/commit/d595fcd7e155d40a04eac95923311d451a58ed02#diff-e0802cf6e5db562b5f045b9917800c82), this should be using ``this.currentDecision``, defined in ``runDecision``.

(/s/Decision/Action)

**Pull request merged**